services/user/ClientData/plugin/src/lib.rs

```diff
@@ -0,0 +1,79 @@
+#[allow(warnings)]
+mod bindings;
+
+mod errors;
+use errors::ErrorType::*;
+
+use bindings::exports::clientdata::plugin::keyvalue::Guest as KeyValue;
+use bindings::host::common::client as Client;
+use bindings::host::common::types as CommonTypes;
+use bindings::wasi::keyvalue as Kv;
+
+struct ClientData;
+
+fn get_sender() -> String {
+    Client::get_sender_app()
+        .app
+        .expect("ClientData interface can only be used from plugins.")
+}
+
+impl KeyValue for ClientData {
+    fn get(key: String) -> Option<Vec<u8>> {
+        let bucket =
+            Kv::store::open(&get_sender()).expect("Failed to open table in keyvalue store");
+
+        // Check that it exists before getting it, because get on a non-existent record
+        // will create an empty record. That's not what we want.
```

Comment:
That's not what https://github.com/WebAssembly/wasi-keyvalue/blob/main/wit/store.wit#L75 says.

Reply:
I kept the wasi-keyvalue implementation maximally aligned with the draft standard put forward by the BCA, but the ClientData plugin is a wrapper around it and it changes that particular behavior intentionally to be more in line with other keyvalue semantics which I think make more sense, like the LocalStorage semantics.

Reply:
AFAICT, neither the key-value spec nor LocalStorage nor our implementation, will create records implicitly on get. What am I missing?

Reply:
... Wow, some mandela effect stuff going on. 
I was sure there was some bizarre semantics in wasi-keyvalue where a record would be created during `get` instead of returning null, but I'm clearly mistaken. I'll remove this.

Reply:
Removed.